crates/sui-graphql-e2e-tests/tests/epoch/pagination.move

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//# init --protocol-version 51 --simulator --accounts C
+
+// TODO: Short term hack to get around indexer epoch issue
+//# create-checkpoint
+
+//# advance-epoch
+
+// TODO: Short term hack to get around indexer epoch issue
+//# create-checkpoint
+
+//# advance-epoch
+
+
+// TODO: Short term hack to get around indexer epoch issue
+//# create-checkpoint
+
+//# advance-epoch
+
+//# create-checkpoint
+
+//# advance-epoch
+
+//# create-checkpoint
+
+//# advance-epoch
+
+//# create-checkpoint
+
+//# advance-epoch
+
+//# run-graphql 
+{
+    epochs(last:2) {
+        pageInfo {
+            hasPreviousPage
+            hasNextPage
+        }
+        nodes {
+            epochId
+        }
+    }
+}
+
+//# run-graphql 
+{
+    epochs(first:3) {
+        pageInfo {
+            hasPreviousPage
+            hasNextPage
+        }
+        nodes {
+            epochId
+        }
+    }
+}
+
+//# run-graphql --cursors {"c":5,"e":2} 
```

Comment:
what about a checkpoint viewed at before 5, and trying to view an epoch beyond that checkpoint, and looking at the full state of epochs per that checkpoint? ie c=3, try to view e=4, c=3, query full `epochs` state starting from epoch 0